nrf_modem/doc/dectphy.rst

```diff
@@ -0,0 +1,296 @@
+
+#. Initialize the Modem library by calling the :c:func:`nrf_modem_init` function. This also turns on the modem SoC.
+#. Register callbacks for DECT PHY operations by calling the :c:func:`nrf_modem_dect_phy_callback_set` function.
+
+Afterwards, the application can initialize the DECT PHY interface by calling the :c:func:`nrf_modem_dect_phy_init` function.
+Upon successful initialization, both the DECT PHY interface and DECT NR+ PHY layer in the modem are ready for operation.
+
+The very first time the DECT PHY interface is initialized the DECT NR+ PHY modem firmware will perform a product locking procedure.
+
+The DECT PHY interface can be de-initialized, which in turn de-initializes the PHY layer in the modem, cancelling all scheduled operations.
+
+.. note::
+   De-initializing the DECT PHY interface does not unset any callbacks set by the application using the :c:func:`nrf_modem_dect_phy_callback_set` function,
+   nor it affects the modem SoC power status (on/off), which is instead controlled by the :c:func:`nrf_modem_init` and :c:func:`nrf_modem_shutdown` functions.
+
+Once de-initialized, the DECT PHY interface can be re-initialized by calling the :c:func:`nrf_modem_dect_phy_init` function only.
+
+Radio states
+============
+
+The radio states have implications on latency and power consumption. During operation, the radio can be in one of several states.
+
+When a radio operation is being executed, the stack is in one of the following states:
+
+* TX active: the radio is transmitting, no reception is ongoing.
+* RX active: the radio is receiving, no transmission is ongoing.
+
+When no radio operations is being executed, the stack is in the following state:
+
+* Radio Idle: the radio is powered on and able to start a radio operation with relatively low latency, at the cost of increased power consumption compared sleep or off states.
+
+Currently, the DECT PHY layer in the DECT NR+ modem firmware does not support deep sleep states.
+
+Scheduling operations
+=====================
+
+The DECT PHY interface allows to schedule radio operations for execution by the scheduler of the DECT NR+ physical layer in the DECT NR+ modem firmware.
+Due to the nature of a radio scheduler, which allows radio operations to be executed at a specific time in the future, all radio operations in the DECT PHY interface are asynchronous and their completion is signalled to the application using callbacks.
+
+All scheduling is final, meaning that it is not possible to unschedule a scheduled operation.
+
+A radio operation may be scheduled to execute a specific time in the future, or immediately, if the radio is not currently executing any other operation.
+
+All radio operations have an application-defined handle which can be used to identify the operation and that is returned by the callback signalling the completion of an operation.
+The operation handle is entirely opaque to all underlying layers, including the DECT PHY interface.
```

Comment:
I might be good already here restrict the usage of handle. In mfw_1.1.0 we will introduce a restriction to handle usage: active operations shall have unique handle values in DECT PHY, i.e. it's not allowed in future to schedule 2 operations to DECT PHY with same handle values. This will simplify later our support for "removing scheduled operations"

Reply:
Also the handle range is restricted later in mfw_1.1.0 to [0, (UINT32_MAX-1)]